content/2023-10-02-W-global-constructors.rst

```diff
@@ -0,0 +1,162 @@
+-Wglobal-constructors
+#####################
+
+:date: 2023-10-02-23
+:category: mozilla
+:lang: en
+:authors: Serge Guelton
+:summary: What's worse than a global variable? A global variable initialized at
+          startup!
+
+
+.. code-block:: c++
+
+    #include <iostream>
+    #include <unordered_set>
+    #include <string>
+
+    const std::unordered_set<std::string> primary_colors = {"red", "green", "blue"};
+
+    int main(int argc, char const ** argv) {
+        char const * color = argc > 1 ? argv[1] : "pink";
+        return primary_colors.count(color) ? 0 : 1;
+    }
+
+Let's compile and run the above snippet:
+
+.. code-block:: shell
+
+   $ clang++ --version
+   clang version 14.0.5 (Fedora 14.0.5-2.fc36)
+   Target: x86_64-redhat-linux-gnu
+   Thread model: posix
+   InstalledDir: /usr/bin
+
+   $ clang++ colors.cpp -O2 -o colors && ./colors green && echo ok
+   ok
+
+Let's be a bit more pedantic and turn all warnings on:
+
+
+.. code-block:: shell
+
+   $ clang++ colors.cpp -Wall -O2 -o colors && ./colors green && echo ok
+   ok
+
+No, I meant really pendantic
```

Comment:
```suggestion
No, I meant really pedantic
```

Reply:
You could even write
```suggestion
No, I meant really pedantic (so that you don't waste your time trying: -pedantic doesn't find anything else)
```